### PR TITLE
Client: Simple message queue to handle unhandled ETH messages during handshake

### PR DESCRIPTION
**Pull request by holgerd77**
Follow-up solution to initial ETH messages being lost, replaces #1236

### CODE REVIEW

File: packages/client/lib/net/peer/peer.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export class Peer extends events.EventEmitter {
   protected transport: string
   protected protocols: Protocol[]
   private _idle: boolean
+  public pooled: boolean = false
```
Comment (ryanio) [inline packages/client/lib/net/peer/peer.ts:49]: PR looks really good and it worked for me on a test run with aluet. I will push a commit to fix the failing tests.

I would like some more clarity on the naming of this `pooled` variable, is there a better name we could use that's more self explanatory? Maybe like "ready" or "handshakeComplete"?

Comment (holgerd77) [reply packages/client/lib/net/peer/peer.ts:49]: Hmm, I like the `pooled` as in "is in the pool", which is actually what this variable describes (it is set to `pooled` when added to the pool and set to `pooled=false` when removed from the pool. This `handshakeComplete` is just one characteristics which comes along, but this variable could also be used for user use cases (not sure if there are any though).

Thanks for dealing with the tests! 😄

Comment (ryanio) [reply packages/client/lib/net/peer/peer.ts:49]: ok sounds good :) I will add a small typedoc to explain that then

**Comment by holgerd77** [comment]
Looks good! @ryanio can you review? 😄

**Review by ryanio — APPROVED** [review]

**Pull request merged**